framework/src/controller/ipc/ipc_server.ts

```diff
@@ -12,46 +12,48 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-// eslint-disable-next-line
-/// <reference path="../../../external_types/pm2-axon/index.d.ts" />
-// eslint-disable-next-line
-/// <reference path="../../../external_types/pm2-axon-rpc/index.d.ts" />
-
-import * as axon from 'pm2-axon';
-import { PubSocket, PullSocket } from 'pm2-axon';
+import { Publisher, Subscriber } from 'zeromq';
 import { IPCSocket } from './ipc_socket';
 
+interface serverSocketPaths {
+	readonly pub: string;
+	readonly sub: string;
+	readonly rpcServer: string;
+}
+
 export class IPCServer extends IPCSocket {
+	private readonly _socketPaths: serverSocketPaths;
+
 	public constructor(options: { socketsDir: string; name: string }) {
 		super(options);
 
-		this.pubSocket = axon.socket('pub', {}) as PubSocket;
-		this.subSocket = axon.socket('pull', {}) as PullSocket;
+		this.pubSocket = new Publisher();
+		this.subSocket = new Subscriber();
+		this._socketPaths = {
```

Comment:
Since we are assigning this value in the constructor we can make it `public readonly socketPaths;` then we don't need the `getter` defined below.